bootstrap/v2/server.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 OpenFaaS Authors
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/*
+
+OpenFaaS API server example:
+
+import bootstrap "github.com/openfaas/faas-provider/bootstrap/v2"
+
+config := bootstrap.ApiServerConfig{
+	ReadTimeout:  cfg.ReadTimeout,
+	WriteTimeout: cfg.WriteTimeout,
+	TCPPort:      &port,
+	EnableHealth: true,
+}
+
+handlers = bootstrap.ApiServerHandlers{
+	FunctionProxy:  handlers.MakeProxy(functionNamespace, config.ReadTimeout),
+	DeleteHandler:  handlers.MakeDeleteHandler(functionNamespace, clientset),
+	DeployHandler:  handlers.MakeDeployHandler(functionNamespace, factory),
+	FunctionReader: handlers.MakeFunctionReader(functionNamespace, clientset),
+	ReplicaReader:  handlers.MakeReplicaReader(functionNamespace, clientset),
+	ReplicaUpdater: handlers.MakeReplicaUpdater(functionNamespace, clientset),
+	UpdateHandler:  handlers.MakeUpdateHandler(functionNamespace, factory),
+	HealthHandler:  handlers.MakeHealthHandler(),
+	InfoHandler:    handlers.MakeInfoHandler(version.BuildVersion(), version.GitCommit),
+	SecretHandler:  handlers.MakeSecretHandler(functionNamespace, clientset),
+}
+
+srv := bootstrap.NewApiServer(config, handlers)
+
+log.Fatal(srv.ListenAndServe())
+
+*/
+
+package v2
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/openfaas/faas-provider/auth/v1"
+)
+
+// ApiServer represents the OpenFaaS API
+type ApiServer struct {
+	Router   *mux.Router
+	handlers ApiServerHandlers
+	config   ApiServerConfig
+}
+
+// NewApiServer returns an OpenFaaS API server
+func NewApiServer(config ApiServerConfig, handlers ApiServerHandlers) *ApiServer {
+	return &ApiServer{
+		config:   config,
+		handlers: handlers,
+		Router:   mux.NewRouter(),
+	}
+}
+
+// ListenAndServe load your handlers into the correct OpenFaaS route spec and starts the API server.
+// This function is blocking.
+func (srv *ApiServer) ListenAndServe() error {
```

Comment:
Now seems like a good time to add graceful shutdown

Reply:
@LucasRoesler Can you do that after we merge this PR?

Nobody will be using the "v2" interface at that time, and it may help stop this PR becoming an "ueber-PR-that-is-never-rebased-or-merged".